test/functorch/test_aotdispatch.py

```diff
@@ -6022,6 +6022,101 @@ def fn(x):
         out = torch.compile(fn, backend="aot_eager", fullgraph=True)(inp)
         self.assertEqual(ref_out, out)
 
+    def test_channels_last_grads_no_force_contiguous_dense(self):
+        class M(torch.nn.Module):
+            def __init__(self) -> None:
+                super().__init__()
+                self.conv = torch.nn.Conv2d(3, 3, 3)
+
+            def forward(self, x, y):
+                z = y + 3
+                y.mul_(2)
+                r = self.conv(x)
+                return r, r.transpose(0, 1), z.view(-1), z.transpose(0, 1)
+
+        m = M()
+        m.to(memory_format=torch.channels_last)
+        m.train()
+
+        def dense_inps():
+            ret = (
+                torch.randn(2, 3, 5, 5, requires_grad=True).to(
+                    memory_format=torch.channels_last
+                ),
+                torch.randn(3, 2, 1, 1, requires_grad=True).to(
+                    memory_format=torch.channels_last
+                ),
+            )
+            [inp.retain_grad() for inp in ret]
+            return ret
+
+        ref_inps = dense_inps()
+        ref_outs = m(*ref_inps)
+        ref_outs[0].sum().backward()
+
+        inps = dense_inps()
+        outs = torch.compile(m, backend="aot_eager", fullgraph=True)(*inps)
+        s = outs[0].sum()
+        with torch.profiler.profile(
+            activities=[torch.profiler.ProfilerActivity.CPU],
+        ) as prof:
+            s.backward()
+
+        contiguous_events = [
+            event for event in prof.events() if event.name == "aten::contiguous"
```

Comment:
hmm yeah, I can't think of a better way to test than what you're doing (running the profiler and making sure there is not actually a contiguous() call at runtime).

Just for sanity, did you confirm that this test fails without your change?

Reply:
also, a comment explaining the purpose of this test would be very appreciated

Reply:
Yes, It fails without changes in AOTD.